frontend/javascripts/router.tsx

```diff
@@ -128,6 +136,37 @@ const SecuredRouteWithErrorBoundary: React.FC<GetComponentProps<typeof SecuredRo
   );
 };
 
+const LegacyLinkDisambiguateErrorView: React.FC<{
+  directoryName: string;
+  organizationId: string;
+}> = ({ directoryName, organizationId }) => {
+  const user = useSelector((state: OxalisState) => state.activeUser);
+  const organizationToSwitchTo = useFetch(
+    async () => {
+      return user
+        ? isDatasetAccessibleBySwitching({ directoryName, organizationId, type: "VIEW" })
```

Comment:
> 1. there is a conversion from legacy to non legacy and afterwards the accessibility check is done

Sadly that's not possible in the current model of the backend. @fm3 what do you think about adjusting the permissions for the disambiguation route to include positive answers in case the user can access the dataset when they would switch into the correct organization?

> 2. the accessibility check can work with the old and new format so that it is only called at one code location

Working with the old format is technically possible but here we have the reverse problem: In case the user opens a URL with the new schema `.../datasets/ds_name-12390u1902u31u2301u20/view` and is not in the orga of the`accessibleBySwitching` route only works on dataset ids, legacy links do not work, and if the route works on organization id and directory name, the new links do not work anymore. Therefore, I'd say both variations must be supported or we do your 1. suggestion

Reply:
fine by me, don’t have a strong opinion here

Reply:
Ok I'll implement 1.